### ISSUE TITLE
Add option group to select field ??

### ISSUE THREAD

**Issue by ravishakya** [themeum/kirki#1120]
### Issue description:

I want to add option group to select field. I want to display posts and pages in the select field. If i dont put optgroup then the user cannot distinguish between which is post and which is pages.  Is it possible to add optgroup tag ??

eg output.

 `<select>`
       `<optgroup label="Posts">`
    `<option value="volvo">Volvo</option>`
    `<option value="saab">Saab</option>`
  `</optgroup>`
 `<optgroup label="Pages">`
   `<option value="mercedes">Mercedes</option>`
    `<option value="audi">Audi</option>`
  `</optgroup>`
`</select>`

**Comment by aristath** [comment]
I'm planning to refactor the select fields soon so I'm adding this one as a feature request...

**Comment by aristath** [comment]
Implemented in 3.0 (develop branch).

Example:
```php
Kirki::add_field( 'my_theme', array(
	'type'        => 'select',
	'settings'    => 'my_setting',
	'label'       => __( 'This is the label', 'my_textdomain' ),
	'section'     => 'my_section',
	'default'     => 'option-1',
	'priority'    => 10,
	'multiple'    => 3,
	'choices'     => array(
		'option-1' => esc_attr__( 'Option 1', 'my_textdomain' ),
		'option-2' => esc_attr__( 'Option 2', 'my_textdomain' ),
		'optgroup1' => array(
			esc_attr__( 'Optgroup 1', 'my_textdomain' ),
			array(
				'option-3' => esc_attr__( 'Option 3', 'my_textdomain' ),
				'option-4' => esc_attr__( 'Option 4', 'my_textdomain' ),
				'option-5' => esc_attr__( 'Option 5', 'my_textdomain' ),
			),
		),
		'optgroup2' => array(
			esc_attr__( 'Optgroup 2', 'my_textdomain' ),
			array(
				'option-6' => esc_attr__( 'Option 6', 'my_textdomain' ),
				'option-7' => esc_attr__( 'Option 7', 'my_textdomain' ),				
			),
		),
		'option-8' => esc_attr__( 'Option 8', 'my_textdomain' ),
	),
) );
```
